public/app/features/alerting/unified/hooks/useAbilities.ts

```diff
@@ -49,53 +58,137 @@ export enum AlertmanagerAction {
+
+  return {
+    // internal (Grafana managed)
+    [AlertingAction.CreateAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleCreate)],
+    [AlertingAction.ViewAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleRead)],
+    [AlertingAction.UpdateAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleUpdate)],
+    [AlertingAction.DeleteAlertRule]: [AlwaysSupported, ctx.hasPermission(AccessControlAction.AlertingRuleDelete)],
+    [AlertingAction.ExportGrafanaManagedRules]: [AlwaysSupported, canExportGMARules],
+
+    // external
+    [AlertingAction.CreateExternalAlertRule]: [
       AlwaysSupported,
       ctx.hasPermission(AccessControlAction.AlertingRuleExternalWrite),
     ],
-    [AlertSourceAction.ViewExternalAlertRule]: [
+    [AlertingAction.ViewExternalAlertRule]: [
       AlwaysSupported,
       ctx.hasPermission(AccessControlAction.AlertingRuleExternalRead),
     ],
-    [AlertSourceAction.UpdateExternalAlertRule]: [
+    [AlertingAction.UpdateExternalAlertRule]: [
       AlwaysSupported,
       ctx.hasPermission(AccessControlAction.AlertingRuleExternalWrite),
     ],
-    [AlertSourceAction.DeleteExternalAlertRule]: [
+    [AlertingAction.DeleteExternalAlertRule]: [
       AlwaysSupported,
       ctx.hasPermission(AccessControlAction.AlertingRuleExternalWrite),
     ],
   };
+};
+
+export const useAlertingAbility = (action: AlertingAction): Ability => {
+  const allAbilities = useAlertingAbilities();
+  return allAbilities[action];
+};
+
+/**
+ * This hook will check if we support the action and have sufficient permissions for it on a single alert rule
+ */
+export function useAlertRuleAbility(rule: CombinedRule, action: AlertRuleAction): Ability {
+  const abilities = useAllAlertRuleAbilities(rule);
+
+  return useMemo(() => {
+    return abilities[action];
+  }, [abilities, action]);
+}
+
+export function useAlertRuleAbilities(rule: CombinedRule, actions: AlertRuleAction[]): Ability[] {
+  const abilities = useAllAlertRuleAbilities(rule);
+
+  return useMemo(() => {
+    return actions.map((action) => abilities[action]);
+  }, [abilities, actions]);
+}
+
+export function useAllAlertRuleAbilities(rule: CombinedRule): Abilities<AlertRuleAction> {
+  const rulesSource = rule.namespace.rulesSource;
+  const rulesSourceName = typeof rulesSource === 'string' ? rulesSource : rulesSource.name;
+
+  const isProvisioned = isGrafanaRulerRule(rule.rulerRule) && Boolean(rule.rulerRule.grafana_alert.provenance);
+  const isFederated = isFederatedRuleGroup(rule.group);
+
+  // if a rule is either provisioned or a federated rule, we don't allow it to be removed or edited
+  const immutableRule = isProvisioned || isFederated;
+
+  // TODO refactor this hook maybe
+  const {
+    isEditable,
+    isRemovable,
+    isRulerAvailable = false,
+    loading,
+  } = useIsRuleEditable(rulesSourceName, rule.rulerRule);
```

Comment:
yes, I think we could extract this permissions logic that are inside this `useIsRuleEditable` and use it here. So, this `useIsRuleEditable` would only return if the rule is editable but don't check any permission.  I think all permissions should be checked in here.